hudi-spark-datasource/hudi-spark/src/test/scala/org/apache/hudi/functional/TestCOWDataSource.scala

```diff
@@ -320,4 +320,21 @@ class TestCOWDataSource extends HoodieClientTestBase {
 
     assertTrue(HoodieDataSourceHelpers.hasNewCommits(fs, basePath, "000"))
   }
+
+  @Test def testWithEmptyInput(): Unit = {
+    val inputDF1 = spark.read.json(spark.sparkContext.parallelize(Seq.empty[String], 1))
+    inputDF1.write.format("org.apache.hudi")
+      .options(commonOpts)
+      .option(DataSourceWriteOptions.OPERATION_OPT_KEY, DataSourceWriteOptions.INSERT_OPERATION_OPT_VAL)
+      .mode(SaveMode.Overwrite)
+      .save(basePath)
+    assertTrue(HoodieDataSourceHelpers.hasNewCommits(fs, basePath, "000"))
```

Comment:
I don't think we should make an empty commit. Empty commits might pollute the timeline, cleaning, compactions e.t.c. and trigger some unexpected behaviors. WDYT?

Reply:
I agree. We may have unexpected bugs for this in the future. 


Why do SparkRDDWriteClient or HoodieTable create a new commit in case of empty dataset (or no record received)?

Reply:
Since this PR is trying to remove the `isEmpty()` check. If we have that check, we won't make an empty commit.

Reply:
Well, it seems that we miss a `return` key word for the `isEmpty` `if-branch` in the origin implement. So I think this PR should not bring new bugs.

Reply:
This seems like a bug introduced here... https://github.com/apache/hudi/pull/1121/files
We should definitely return if the incoming record is empty. cc: @vinothchandar WDYT?

Reply:
I might be mistaken here. I see delta streamer and Flink could also make an empty commit. If empty commits are not considering as a bug, then this should be fine.
cc: @bvaradar @vinothchandar can you guys make the call here?

Reply:
I too vote for not making an empty commit if its empty. Will wait to hear from Vinoth/balaji

Reply:
Hi @nsivabalan ,I also agree that not making an empty commit. Because it would produce many useless commit files for streaming job when the source of the streaming is in the business trough. And I have update the code to skip the empty commit.

Reply:
The empty commits has skip now. And the test case has fixed @nsivabalan @garyli1019 . Please take a look when you have time. Thanks~